.github/workflows/ci.yml

```diff
@@ -0,0 +1,47 @@
+name: CI
+on:
+  push:
+    branches: [develop]
+  pull_request:
+
+env:
+  CXXFLAGS: "-fprofile-arcs -ftest-coverage"
+
+jobs:
+  test:
+    name: "Python ${{ matrix.python-version }} / ${{ matrix.os }}"
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        python-version: ["2.7", "3.8"]
+        os: [ubuntu-latest]
+        compiler: [gcc]
+    env:
+      CC: ${{ matrix.compiler }}
+      PY2: ${{ startsWith(matrix.python-version, 2) && 'on' || 'off' }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install dependencies
+        run: |
+          sudo apt-get update -qq
+          sudo apt-get install -qq libboost-all-dev swig libhdf5-serial-dev libeigen3-dev
+      - name: Set up python
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+          architecture: 'x64'
+      - name: Set up git
+        run: ./setup_git.py
+      - name: Build IMP
+        run: |
+          mkdir ../build
+          cd ../build
+          cmake ../imp -DUSE_PYTHON2=${{ env.PY2 }} -DCMAKE_CXX_FLAGS="${{ env.CXXFLAGS }}"
+          make -k -j 2
+      - name: Run tests
+        run: ctest -j 2 --output-on-failure
+      - name: Upload coverage to Codecov
```

Comment:
I would skip expensive tests here, so that things run in a reasonable time (and frankly, expensive tests shouldn't be relied upon for coverage anyway, since they're basically unusable for a debug build). Add `-E expensive` to the  `ctest` invocation.  Also, are you sure you don't need to `cd ../build` here too? That might be why it's not finding any tests.

Reply:
Note also that I ran tests directly with `nosetests --with-coverage`. (See the PMI repo for the full setup; there's a bit of a hack there since IMP tests weren't written with nose in mind.) Not sure if ctest can collect Python coverage in a format that codecov will accept.

Reply:
> I would skip expensive tests here
Good point, will do.

> Also, are you sure you don't need to `cd ../build` here too? That might be why it's not finding any tests.

I can try that. It's curious that the Travis build doesn't need a `cd ../build` to find the tests.

Reply:
> Note also that I ran tests directly with `nosetests --with-coverage`.

Do you mean run `nosetests` in place of `ctest` or in addition to? (i.e. will `nosetest` get c++ coverage as well?)

Reply:
I run it in place of `ctest`. It won't run the C++ tests but the Python tests will still exercise the underlying C++ code and collect coverage. The vast majority of IMP tests are Python.

Reply:
> > Also, are you sure you don't need to `cd ../build` here too? That might be why it's not finding any tests.
> 
> I can try that. It's curious that the Travis build doesn't need a `cd ../build` to find the tests.

It turns out that any directory change in one step is undone in the next, so mystery solved.